test/collections.js

```diff
@@ -317,6 +317,10 @@
     assert.deepEqual(_.filter(list, {b: 2}), [{a: 1, b: 2}, {a: 2, b: 2}]);
     assert.deepEqual(_.filter(list, {}), list, 'Empty object accepts all items');
     assert.deepEqual(_(list).filter({}), list, 'OO-filter');
+
+    // Works with RegExps
+    var stringArray = ['cat', 'dog', 'catdog'];
+    assert.deepEqual(_.filter(stringArray, /cat/), ['cat', 'catdog']);
```

Comment:
In JavaScript arrays can have mixed type.  I'm guessing we want:

```javascript
_.filter([123, "123", {a: 1}, []], /23/); // [123, "123"]
```

Do you agree?